ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/pull-queries-against-materialized-table-table-joins.json

```diff
@@ -0,0 +1,466 @@
+        {"query": [
+          {"header":{"schema":"`ID` BIGINT KEY, `T_ID` BIGINT, `TT_ID` BIGINT, `NAME` STRING, `VALUE` BIGINT, `F1` STRING, `F2` BIGINT"}},
+          {"row":{"columns":[1, 1, 1, "bar", 99, "a", 10]}},
+          {"row":{"columns":[15, 0, 15, "" , 0, "c", 20]}},
+          {"row":{"columns":[10, 10, 0, "100", 5, "", 0]}}
+        ]}
+      ]
+    },
+    {
+      "name": "table join pipeline",
+      "format": ["JSON"],
+      "statements": [
+        "CREATE TABLE TEST (ID BIGINT PRIMARY KEY, NAME varchar, VALUE bigint) WITH (kafka_topic='left_topic', value_format='{FORMAT}');",
+        "CREATE TABLE TEST_TABLE (ID BIGINT PRIMARY KEY, F1 varchar, F2 bigint) WITH (kafka_topic='right_topic', value_format='{FORMAT}');",
+        "CREATE TABLE TEST_TABLE_2 (ID BIGINT PRIMARY KEY, F3 varchar) WITH (kafka_topic='right_topic_2', value_format='{FORMAT}');",
+        "CREATE TABLE INNER_JOIN WITH (PARTITIONS=4) as SELECT t.id AS ID, name, value, f1, f2 FROM test t join TEST_TABLE tt on t.id = tt.id;",
+        "CREATE TABLE INNER_JOIN_2 AS SELECT tt.id, name, f1, f3 FROM inner_join tt join TEST_TABLE_2 t ON t.id = tt.id;",
+        "SELECT * FROM INNER_JOIN_2;"
+      ],
+      "topics": [
+        {
+          "name": "INNER_JOIN",
+          "format": "JSON",
+          "partitions": 4
+        }
+      ],
+      "inputs": [
+        {"topic": "INNER_JOIN", "key": 0, "value": {"NAME": "X", "VALUE": 0, "F1": "yo dawg", "F2": 50}, "timestamp": 0},
+        {"topic": "right_topic_2", "key": 0, "value": {"F3": "I heard you like joins"}, "timestamp": 10000},
+        {"topic": "INNER_JOIN", "key": 100, "value": {"NAME": "X", "VALUE": 0, "F1": "KSQL has table-table joins", "F2": 50}, "timestamp": 15000},
+        {"topic": "right_topic_2", "key": 100, "value": {"F3": "so now you can join your join"}, "timestamp": 20000}
+      ],
+      "responses": [
+        {"admin": {"@type": "currentStatus"}},
+        {"admin": {"@type": "currentStatus"}},
+        {"admin": {"@type": "currentStatus"}},
+        {"admin": {"@type": "currentStatus"}},
+        {"admin": {"@type": "currentStatus"}},
+        {"query": [
+          {"header":{"schema":"`TT_ID` BIGINT KEY, `NAME` STRING, `F1` STRING, `F3` STRING"}},
+          {"row":{"columns":[100, "X", "KSQL has table-table joins", "so now you can join your join"]}},
+          {"row":{"columns":[0, "X", "yo dawg", "I heard you like joins"]}}
+
+        ]}
+      ]
+    },
+    {
+      "name": "table table join with where clause",
+      "statements": [
+        "CREATE TABLE TEST (ID BIGINT PRIMARY KEY, NAME varchar, VALUE bigint) WITH (kafka_topic='left_topic', value_format='JSON');",
+        "CREATE TABLE TEST_TABLE (ID BIGINT PRIMARY KEY, F1 varchar, F2 bigint) WITH (kafka_topic='right_topic', value_format='JSON');",
+        "CREATE TABLE OUTPUT as SELECT t.id, name, tt.f1, f2 FROM test t JOIN test_table tt ON t.id = tt.id WHERE t.value > 10 AND tt.f2 > 5;",
```

Comment:
Is `table table inner join - filter in topology` subsumed by this test?

Reply:
yes, good catch